clippy_lints/src/manual_unwrap_or_default.rs

```diff
@@ -148,6 +148,42 @@ fn handle<'tcx>(cx: &LateContext<'tcx>, if_let_or_match: IfLetOrMatch<'tcx>, exp
         && is_default_equivalent(cx, body_none)
         && let Some(receiver) = Sugg::hir_opt(cx, condition).map(Sugg::maybe_par)
     {
+        // We check if the expression is not a None variant
+        if let Some(none_def_id) = cx.tcx.lang_items().option_none_variant() {
+            if let ExprKind::Path(QPath::Resolved(_, path)) = &condition.kind {
+                if let Some(def_id) = path.res.opt_def_id() {
+                    if cx.tcx.parent(def_id) == none_def_id {
+                        return span_lint_and_sugg(
+                            cx,
+                            MANUAL_UNWRAP_OR_DEFAULT,
+                            expr.span,
+                            format!("{expr_name} can be simplified with `.unwrap_or_default()`"),
+                            "replace it with",
+                            format!("{receiver}::</* Type */>.unwrap_or_default()"),
```

Comment:
You can print any `Ty` which will print the type's full name. The following should work:

```rust
if let ty::Adt(_, args) = cx.typeck_results().expr_ty(expr).kind()
  && let Some(arg) = args.first()
  && let Some(arg_ty) = arg.as_type()
{
  format!("{receiver}::<{}>.unwrap_or_default()", arg_ty)
}
```